runtime/src/accounts_db.rs

```diff
@@ -4515,6 +4524,14 @@ impl AccountsDb {
         self.flush_slot_cache(slot, None::<&mut fn(&_, &_) -> bool>);
     }
 
+    /// true if write cache is too big or has too many slots
+    fn should_aggressively_flush_cache(&self) -> bool {
+        self.write_cache_limit_bytes
+            .map(|write_cache_limit_bytes| self.accounts_cache.size() >= write_cache_limit_bytes)
+            .unwrap_or_default()
+            || self.accounts_cache.num_slots() > MAX_CACHE_SLOTS
```

Comment:
If we're going with the size based approach, we can probably get rid of the `MAX_CACHE_SLOTS` check